### PR TITLE
add: recover tests & fixes

### PR DESCRIPTION
**Pull request by ilinzweilin**

### CODE REVIEW

File: src/mgr.sol [hunk 1]
```diff
@@ -300,11 +302,12 @@ contract TinlakeManager {
...
     function recover(uint256 endEpoch) public {
         require(!glad, "TinlakeMgr/not-written-off");
-
+ 
         (uint256 recovered, , ,) = pool.disburse(endEpoch);
         uint256 payBack;
         if (end.debt() == 0) {
             payBack = min(recovered, tab / RAY);
+            dai.approve(address(daiJoin), payBack);
             daiJoin.join(address(vow), payBack);
```
Comment (xmxanuel) [inline src/mgr.sol]:
```suggestion
            daiJoin.join(vow, payBack);
```

File: src/test/mgr-unit.t.sol [hunk 2]
```diff
@@ -488,10 +519,27 @@ contract TinlakeManagerUnitTest is DSTest, DSMath {
...
 
+    function testFailRecoverisGlad(uint128 wad) public {
+        recover(wad, 1);
+    }
+
+    function testRecoverGlobalSettlement(uint128 wad) public {
+        if (ceiling < wad) return; // amount has to be below ceiling
+        testCull(wad);
+        cage();
+        recover(wad, 1);
```
Comment (xmxanuel) [inline src/test/mgr-unit.t.sol:567]: maybe we can add here:
```
assertEq(dai.balanceOf(self), wad);
```

**Review by xmxanuel — APPROVED** [review]
LGTM

File: src/test/mgr-unit.t.sol (hunk 2 above)
Comment (ilinzweilin) [reply src/test/mgr-unit.t.sol:567]: the assertions are in the recover function

**Pull request merged**